lasagne/layers/merge.py

```diff
@@ -11,7 +11,143 @@
 ]
 
 
-class ConcatLayer(MergeLayer):
+class MergeCropLayer(MergeLayer):
+    """
+    This abstract base class class adds cropping to MergeLayer.
+
+    Cropping takes a sequence of inputs and crops them per-axis in order to
+    ensure that their sizes are consistent so that they can be combined
+    in an element-wise fashion. When cropping is enabled - layers that
+    support it take a cropping argument to the constructor - each
+    axis has a cropping mode.  If cropping is enabled for a specific axis,
+    the minimum size in that axis of all inputs is computed, and all
+    inputs are cropped to that size.
+
+    `CROP_NONE` (`None`): this axis is not cropped, inputs are unchanged in
+    this axis
+
+    `CROP_LOWER` (`'lower'`): inputs are cropped choosing the lower portion
+    in this axis (`a[:crop_size, ...]`)
+
+    `CROP_UPPER` (`'lower'`): inputs are cropped choosing the upper portion
+    in this axis (`a[-crop_size:, ...]`)
+
+    `CROP_CENTER` (`'center'`): inputs are cropped choosing the central
+    portion in this axis (`a[offset:offset+crop_size, ...]` where
+    `offset = (a.shape[0]-crop_size)//2)
+
+    For example, given three inputs, whose shapes are:
+    `a.shape == (1, 2, 3, 4)`
+    `b.shape == (5, 4, 4, 2)`
+    `c.shape == (7, 1, 8, 9)`
+
+    with crop modes `[CROP_NONE, CROP_LOWER, CROP_CENTER, CROP_UPPER]`
+
+    They will be left as is in axis 0 and cropped in all others,
+    choosing the lower, center and upper portions:
+
+    a[:, :1, :3, -2:]   # Choose all, lower 1 element, 3 central (all)
+                        # and upper 2
+    b[:, :1, :3, -2:]   # Choose all, lower 1 element, 3 central starting at 0
+                        # and upper 2 (all)
+    c[:, :1, 2:5:, -2:] # Choose all, lower 1 element (all),
+                        # 3 central starting at 2 and upper 2 (all)
```

Comment:
Nice explanation!  But could benefit from being turned into a runnable doctest, to make sure all the numbers are correct, too.